thirdparty/runtime_environments.go

```diff
@@ -58,3 +64,38 @@ func (c *RuntimeEnvironmentsClient) getImageNames(ctx context.Context) ([]string
 	}
 	return filteredImages, nil
 }
+
+// getOSInfo returns a list of operating system changes (name and version) from the corresponding AMI id.
+func (c *RuntimeEnvironmentsClient) getOSInfo(ctx context.Context, amiId string, page string, limit string) ([]OSInfo, error) {
+	apiURL := fmt.Sprintf("%s/rest/api/v1/image", c.BaseURL)
+	params := url.Values{}
+	params.Set("ami", amiId)
+	params.Set("page", page)
+	params.Set("limit", limit)
+	params.Set("type", "OS")
+	reqURL, err := url.Parse(apiURL)
+	if err != nil {
+		return nil, errors.Wrap(err, "parsing API url")
+	}
+	reqURL.RawQuery = params.Encode()
+	request, err := http.NewRequestWithContext(ctx, http.MethodGet, reqURL.String(), nil)
```

Comment:
I think this might be a more straightforward way to write the request:
```suggestion
	params := url.Values{}
	params.Set("ami", amiId)
	params.Set("page", page)
	params.Set("limit", limit)
	params.Set("type", "OS")

	apiURL := fmt.Sprintf("%s/rest/api/v1/image?%s", c.BaseURL, params.Encode())

	request, err := http.NewRequestWithContext(ctx, http.MethodGet, apiURL, nil)
```